robocop/checkers/naming.py

```diff
@@ -56,14 +56,9 @@ def check_if_char_in_node_name(self, node, name_of_node, is_keyword=False):
         index = 0
         while index < len(node.name):
             # skip variables
-            i = 0
-            while variables and i < len(variables):
-                start, stop = variables[i]
-                if start <= index < stop:
-                    index += stop - start
-                    variables.pop(i)
-                    break
-                i += 1
+            if variables and variables[0][0] == index:
```

Comment:
Let's assume that I believe you that this works correctly :)

Reply:
Thats not the matter of belief if you have good enough tests!  But yeah lets hope it will fix this issue for good.